crates/turbopack-ecmascript/src/references/esm/url.rs

```diff
@@ -25,6 +25,20 @@ use crate::{
     utils::module_id_to_lit,
 };
 
+/// Determines how to treat `new URL(...)` rewrites.
+/// This allows to construct url depends on the different building context,
+/// e.g. SSR, CSR, or Node.js.
+#[turbo_tasks::value(shared)]
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, Hash)]
+pub enum UrlRewriteBehavior {
+    /// Omits base, resulting in a relative URL.
```

Comment:
What environment would you use a relative URL?

Reply:
https://vercel.slack.com/archives/C03EWR7LGEN/p1699297737138929?thread_ts=1699294919.660049&cid=C03EWR7LGEN

Instead of rewriting every hostname, webpack seems to handle it by having relative url instead. As noted in PR body we have a config for it as well.

> In next.js, depends on the compilation context it specifies how new URL is being treated (https://github.com/vercel/next.js/blob/canary/packages/next/src/build/webpack-config.ts#L1384 / https://github.com/vercel/next.js/blob/canary/packages/next/src/build/webpack-config.ts#L1861)